bob-backend/src/pearl/group.rs

```diff
@@ -32,7 +33,13 @@ impl Group {
         dump_sem: Arc<Semaphore>,
     ) -> Self {
         Self {
-            holders: Arc::new(RwLock::new(vec![])),
+            holders: Arc::new(RwLock::new(HierarchicalFilters::<
```

Comment:
Too complex type, I suggest to create your own `type` alias for this. And you are using it several times in this mod.